### ISSUE TITLE
PG shows warnings without indication of the code they belong to

### ISSUE THREAD

**Issue by RalfJung** [ProofGeneral/PG#548]
When running a Coq file produces warnings, PG will show these warnings in the bottom-right window, but there is no way (that I could find) to figure out which lines the warning are talking about. To reproduce, run this script with Coq 8.13:
```
Class Test := test : True.

Lemma foo : Test.
Proof. exact I. Qed.

Hint Resolve foo.

Lemma bar : 0 = 0.
Proof. exact eq_refl. Qed.
```
This shows
```
Warning: Adding and removing hints in the core database implicitly is deprecated. Please specify a
hint database. [implicit-core-hint-db,deprecated]
Warning: The default value for hint locality is currently "local" in a section and "global"
otherwise, but is scheduled to change in a future release. For the time being, adding hints outside
of sections without specifying an explicit locality is therefore deprecated. It is recommended to
use "export" whenever possible. [deprecated-hint-without-locality,deprecated]
```
Now in this case there's just a single `Hint`, but in a large file it can be near impossible to figure out what the warning is talking about.

**Comment by hendriktews** [comment]
I agree this behavior is not optimal, but I am not sure, how
important it is to fix it. So how big of an issues is this? You
can always see get the line numbers when you run coqc. Also, when
you write new code it is not an issue, because than you are slow
enough and see where the warning belongs.

It's pretty clear why this happens: coqtop gets fed from stdin,
there are no line numbers. It's impossible to fix this
completely: Imagine you have a command spread over 10 lines, then
coqtop will only tell you there is a warning somewhere in these
10 lines, but not the prices line. An approximate fix would not
be too complicated: When processing the response, the
span/overlay, which contains position information, is not too far
away.

**Comment by cpitclaudel** [comment]
>  It's impossible to fix this
> completely: Imagine you have a command spread over 10 lines, then
> coqtop will only tell you there is a warning somewhere in these
> 10 lines, but not the prices line.

But it does tell you the precise offset, and we report it correctly for errors, just not for warnings.

**Comment by hendriktews** [comment]
Right, there are character numbers in the output of coqtop, merging them with the position of the span/overlay should give the precise location.

**Comment by RalfJung** [comment]
> I agree this behavior is not optimal, but I am not sure, how
important it is to fix it. So how big of an issues is this? You
can always see get the line numbers when you run coqc. Also, when
you write new code it is not an issue, because than you are slow
enough and see where the warning belongs.

It often happens that new Coq versions show new warnings; in that case the current behavior means PG is just not helpful at all (beyond knowing "somewhere in this file something is wrong" and then having to go back and forth between terminal and emacs to figure out what and where).

**Comment by hendriktews** [comment]
Terminal? I do M-x compile and then C-x ` to step through the
warnings, all without leaving emacs. True, Proof General is not
helpful here, but I am not sure, any more help is needed. If it
is something like replacing omega with lia, then M-x
tags-query-replace does also help a lot.

**Comment by RalfJung** [comment]
I am not an emacs expert, so I never managed to achieve that level of integration.^^

**Comment by RalfJung** [comment]
Would be really nice if PG could at least show the line number for each warning. :)
Currently it's kind of tedious to figure out what all the new warning one gets after a Rocq update refer to.
